Cubical/Categories/Displayed/AdjointToReindex.agda

```diff
@@ -0,0 +1,173 @@
+  variable
+    ℓC ℓC' ℓCᴰ ℓCᴰ' ℓD ℓD' ℓDᴰ ℓDᴰ' ℓE ℓE' ℓEᴰ ℓEᴰ' : Level
+
+open Categoryᴰ
+
+module _
+  {C : Category ℓC ℓC'} {D : Category ℓD ℓD'}
+  (Cᴰ : Categoryᴰ C ℓDᴰ ℓDᴰ') (F : Functor C D)
+  where
+
+  open Category
+  open Functor F
+  private
+    module Cᴰ = Categoryᴰ Cᴰ
+    open Iso
+
+    hom-path : ∀ {A B A' B'} (p : A ≡ A') (q : B ≡ B') →
+      (D [ A , B ]) ≡ (D [ A' , B' ])
+    hom-path p q = cong₂ (λ a b → D [ a , b ]) p q
+
+    hom-pathP : ∀ {A B A' B'} (p : A ≡ A') (q : B ≡ B') →
+                (f : D [ A , B ]) → (f' : D [ A' , B' ]) →
+                Type ℓD'
+    hom-pathP p q f f' = PathP (λ i → hom-path p q i) f f'
+
+    isPropHomP : ∀ {A B A' B'} (p : A ≡ A') (q : B ≡ B') →
+                (f : D [ A , B ]) → (f' : D [ A' , B' ]) →
+                isProp (hom-pathP p q f f')
+    isPropHomP pdom pcod f f'' x y =
+      isoFunInjective (PathPIsoPath _ _ _) x y fromPathPeq
+      where
+      fromPathPeq : fromPathP x ≡ fromPathP y
+      fromPathPeq = D .isSetHom _ _ (fromPathP x) (fromPathP y)
+
+  ∃F : Categoryᴰ D (ℓ-max (ℓ-max ℓC ℓD) ℓDᴰ) (ℓ-max (ℓ-max ℓC' ℓD') ℓDᴰ')
+  ∃F .ob[_] d = Σ[ c ∈ C .ob ] (F-ob c ≡ d) × Cᴰ.ob[ c ]
+  ∃F .Hom[_][_,_] f (c , p , x) (c' , p' , x') =
+    Σ[ g ∈ C [ c , c' ] ] hom-pathP p p' (F-hom g) f × Cᴰ.Hom[ g ][ x , x' ]
+  ∃F .idᴰ {d} {c , p , x} =
+    (C .id) ,
+    (F-id ◁ (cong (λ v → D .id {v}) p)) ,
+    Cᴰ .idᴰ
+  ∃F ._⋆ᴰ_ (g , p , gᴰ) (h , q , hᴰ) =
+      g ⋆⟨ C ⟩ h ,
+      F-seq g h ◁ (λ i → p i ⋆⟨ D ⟩ q i) ,
+      (Cᴰ._⋆ᴰ_ gᴰ hᴰ)
+  ⋆IdLᴰ ∃F {d}{d'}{g}{c , pc , xc}{c' , pc' , xc'} (f , p , x) =
+    λ i →
+      C .⋆IdL f i ,
+      isProp→PathP
+        (λ i → isPropHomP pc pc' (F-hom (C .⋆IdL f i)) (D .⋆IdL g i))
+        ((F-seq (C .id {c}) f ◁
+          (λ i₁ → seq' D ((F-id {c} ◁ (λ i₂ → D .id {pc i₂})) i₁) (p i₁))))
+        p i ,
+      Cᴰ .⋆IdLᴰ x i
+  ⋆IdRᴰ ∃F {d}{d'}{g}{c , pc , xc}{c' , pc' , xc'} (f , p , x) =
+    λ i →
+      C .⋆IdR f i ,
+      isProp→PathP
+        (λ i → isPropHomP pc pc' (F-hom (C .⋆IdR f i)) (D .⋆IdR g i))
+        (F-seq f (C .id {c'}) ◁
+          (λ i₁ → seq' D (p i₁) ((F-id {c' }◁ (λ i₂ → D .id {pc' i₂})) i₁)))
+        p i ,
+      Cᴰ .⋆IdRᴰ x i
+  ⋆Assocᴰ ∃F {xᴰ = xᴰ}{yᴰ = yᴰ}{zᴰ = zᴰ}{wᴰ = wᴰ}
+    (f , pf , fᴰ)
+    (g , pg , gᴰ)
+    (h , ph , hᴰ) =
+    λ i →
+      C .⋆Assoc f g h i ,
+      isProp→PathP
+        (λ j → isPropHomP
+          (xᴰ .snd .fst) (wᴰ .snd .fst)
+          (F-hom (C .⋆Assoc f g h j)) (D .⋆Assoc _ _ _ j))
+        (F-seq (seq' C f g) h ◁
+          (λ i₁ → seq' D
+            ((F-seq f g ◁ (λ i₂ → seq' D (pf i₂) (pg i₂))) i₁) (ph i₁)))
+        (F-seq f (seq' C g h) ◁
+          (λ i₁ → seq' D
+          (pf i₁) ((F-seq g h ◁ (λ i₂ → seq' D (pg i₂) (ph i₂))) i₁)))
+        i ,
+      Cᴰ .⋆Assocᴰ fᴰ gᴰ hᴰ i
+  isSetHomᴰ ∃F {d}{d'}{f}{u}{v} =
+    isSetΣ (C .isSetHom)
+      λ g → isSet×
+        (isProp→isSet (isPropHomP _ _ _ _))
+        (Cᴰ .isSetHomᴰ)
+
+-- Examples of ∃F instantiated
```

Comment:
I thought I'd at least try to demonstrate to y'all that this construction does indeed generalize what we'd discussed in #1106 

These examples can probably be removed though